File: src/main/java/me/fru1t/streamtools/Window.java
package me.fru1t.streamtools;

/**
 * Simple java object containing data for windows within the main menu. This is used to store
 * settings in a Json format.
 */
public class Window {
    public String controllerClass;
    public String settingsJson;

    public double stageWidth;
    public double stageHeight;
    public double stageX;
    public double stageY;
    public String title;
    public boolean isVisible;
}

File: src/main/kotlin/me/fru1t/streamtools/controller/mainmenu/Window.kt
package me.fru1t.streamtools.controller.mainmenu

/** Used to store window data for saving and loading in JSON. */
data class Window(
  var controllerClass: String? = null,
  var settingsJson: String? = null,

  var stageWidth: Double = 0.toDouble(),
  var stageHeight: Double = 0.toDouble(),
  var stageX: Double = 0.toDouble(),
  var stageY: Double = 0.toDouble(),
  var title: String? = null,
  var isVisible: Boolean = false
)
